Optionally accept RELATED_IMAGE_KEYCLOAK as the operand coordinate

ASzc opened the pull request:
This changes the quarkus configuration of the operator so that if the RELATED_IMAGE_KEYCLOAK environment variable is present, it will be accepted over the default, or the existing configuration environment variable of OPERATOR_KEYCLOAK_IMAGE

Closes #24017

<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

ASzc committed:
Optionally accept RELATED_IMAGE_KEYCLOAK as the operand coordinate

This changes the quarkus configuration of the operator so that if the
RELATED_IMAGE_KEYCLOAK environment variable is present, it will be
accepted over the default value

Closes #24017

shawkins reviewed (approved):
LGTM, thanks @ASzc

vmuzikar commented on operator/src/main/resources/application.properties:
Do we also need to change the name of the env var here?
https://github.com/keycloak/keycloak/blob/5cab2a58d7a857e152f2b6d667b13d033bf88c7f/operator/src/main/resources/application.properties#L11

I.e. instead that have
```
quarkus.kubernetes.env.vars.related-image-keycloak=${operator.keycloak.image}
```

This variable affects also the env var name in resulting CSV.

vmuzikar replied:
It does not affect our config. The actual config option name will still remain `operator.keycloak.image`. I'm talking only about using the new env var shortcut (`RELATED_IMAGE_KEYCLOAK`) in the generated K8s resources and CSV. If I understand correctly, this is needed for the prod process.

shawkins replied:
> It does not affect our config.

Sorry, I didn't think through what the full affect would be.  The property:

operator.keycloak.image=${RELATED_IMAGE_KEYCLOAK:quay.io/keycloak/keycloak:nightly} 

will map the RELATED_IMAGE_KEYCLOAK env to our Config.  

>  If I understand correctly, this is needed for the prod process.

Yes, it does look like it needs to already be present in the CSV.

ASzc replied:
My plan was to change the name name in the CSV in my build script. That way there's no breaking change introduced here, and keycloak users can carry on as normal while this unfortunate detail of OSBS is handled.

vmuzikar replied:
It's not a breaking change, old env var name still works. I'd like to have up- and downstream aligned as much as possible.

ASzc replied:
Oh, ok! It does make sense to keep them closer together. Yeah, so if you want I can include that change in this PR too.

vmuzikar replied:
@ASzc Yes, please. :)

ASzc replied:
Done

mabartos commented on operator/src/main/resources/application.properties:
```diff
@@ -4,7 +4,7 @@ quarkus.docker.dockerfile-jvm-path=Dockerfile
 
 # Operator config
-operator.keycloak.image=quay.io/keycloak/keycloak:nightly
+operator.keycloak.image=${RELATED_IMAGE_KEYCLOAK:quay.io/keycloak/keycloak:nightly}
```
```suggestion
operator.keycloak.image=${RELATED_KEYCLOAK_IMAGE:quay.io/keycloak/keycloak:nightly}
```
Nitpick: Would be good to keep some sort of convention to the naming. WDYT?

shawkins replied:
RELATED_IMAGE_something is the required convention here.

mabartos replied:
@shawkins Ah, OK, thanks for the explanation. Ignore the suggestion.

mabartos reviewed (commented):
@ASzc I've added one nitpick.

I assume a normal user will not have any use case to leverage the functionality as it's mainly related to the internal build process, right? I mean, the config does not have to be documented, right?

ASzc commented:
Updated and rebased, I've filed a backport PR too, since this needs to land in 22.x today, ideally.

https://github.com/keycloak/keycloak/pull/24062

vmuzikar reviewed (approved):
@ASzc LGTM, thank you!
